packages/transactions/src/coders/transaction.ts

```diff
@@ -291,7 +294,62 @@ export class TransactionCreateCoder extends Coder<TransactionCreate, Transaction
   }
 }
 
-export type Transaction = TransactionScript | TransactionCreate;
```

Comment:
Could you elaborate on why this was removed? I think we can do  `TransactionScript | TransactionCreate | TransactionMint`, but I may be missing something

Reply:
it wasn't removed, it's just ahead in the code

Reply:
`TransactionScript | TransactionCreate | TransactionMint` was the initial idea, but then it breaks a lot of transactions because of small number of common properties.

then I did a refactor on typing stuff for `Transaction` which now accepts a generic informing transaction type

Reply:
@QuinnLee you can check the changed tests/files to understand better how it's working now